aws/resource_aws_iam_role.go

```diff
@@ -347,30 +356,21 @@ func resourceAwsIamRoleDelete(d *schema.ResourceData, meta interface{}) error {
 		return nil
 	})
 	if isResourceTimeoutError(err) {
-		_, err = iamconn.DeleteRole(deleteRoleInput)
-	}
-
-	if isAWSErr(err, iam.ErrCodeNoSuchEntityException, "") {
-		return nil
+		_, err = conn.DeleteRole(deleteRoleInput)
 	}
 
-	if err != nil {
-		return fmt.Errorf("Error deleting IAM Role (%s): %s", d.Id(), err)
-	}
-	return nil
+	return err
 }
 
 func deleteAwsIamRoleInstanceProfiles(conn *iam.IAM, rolename string) error {
 	resp, err := conn.ListInstanceProfilesForRole(&iam.ListInstanceProfilesForRoleInput{
 		RoleName: aws.String(rolename),
 	})
-
-	if isAWSErr(err, iam.ErrCodeNoSuchEntityException, "") {
+	if tfawserr.ErrCodeEquals(err, iam.ErrCodeNoSuchEntityException) {
 		return nil
 	}
-
 	if err != nil {
-		return fmt.Errorf("Error listing Profiles for IAM Role (%s) when trying to delete: %s", rolename, err)
```

Comment:
just curious if we'll be missing this fine-grain info if we're just returning the `err` or is the API generally pretty good about returning the error with enough info for practitioners to debug?

Reply:
This error case was several levels of wrapping, so we'd get something noisy like

> error deleting IAM Role (\<rolename>) instance profiles: Error listing Profiles for IAM Role (\<rolename>) when trying to delete: \<error>